app/Http/Controllers/Service/Admin/DeliveriesController.php

```diff
@@ -36,8 +77,110 @@ public function create()
         return view('service.deliveries.create', compact('sponsors'));
     }
 
-    public function store()
+
+    /**
+     * @param AdminNewDeliveryRequest $request
+     * @return RedirectResponse
+     * @throws Throwable
+     */
+    public function store(AdminNewDeliveryRequest $request)
     {
-        // implement store
+        // Get centre
+        $centre = Centre::findOrFail($request->input('centre'));
+
+        $startCode = $request->input('voucher-start');
+        $endCode = $request->input('voucher-end');
+
+        // Break the codes up.
+        $start = Voucher::splitShortcodeNumeric($startCode);
+        $start["number"] = intval($start["number"]);
+
+        $end = Voucher::splitShortcodeNumeric($endCode);
+        $end["number"] = intval($end["number"]);
+
+        // Check the voucher range is clear to be delivered.
+        if (!self::rangeIsUndelivered($start["number"], $end["number"], $start["shortcode"])) {
+            // Whoops! Some of the vouchers may have been delivered
+            return redirect()
+                ->route('admin.deliveries.create')
+                ->withInput()
+                ->with('error_message', 'The voucher range given contains some vouchers that have already been delivered.');
+        };
+
+        // Create delivery or roll back
+        try {
+            DB::transaction(function () use ($start, $end, $centre, $request) {
+
+                $delivery = Delivery::create([
+                    'centre_id' => $centre->id,
+                    'range' => $start[0] . " - " . $end[0],
+                    'dispatched_at' => Carbon::createFromFormat('Y-m-d', $request->input('date-sent')),
+                ]);
+
+                $now_time = $delivery->created_at;
+                $user_id = auth()->id();
+                $user_type = class_basename(auth()->user());
+
+                // Bulk update VoucherStates for speed.
+                Voucher::select('id')
+                    ->where('currentState', 'printed')
+                    ->whereNull('delivery_id')
+                    ->whereBetween(
+                        DB::raw("cast(replace(code, '{$start["shortcode"]}', '') as signed)"),
```

Comment:
This allows for injection with an input similar to: `SELECT/**/*/**/FROM/**/users12356789`

Reply:
The voucher the shortcode is split from is validated as real before the controller function is hit. If it doesn't exist it's going to get caught earlier than this.